### PR TITLE
Added kops bucket

### PR DESCRIPTION
**Pull request by lewismarshall**

### CODE REVIEW

File: s3.tf [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+  acl    = "private"
+
+  tags = "${merge(var.tags,
+    map("Name", format("%s.%s", var.environment, var.dns_zone)),
+    map("Env", var.environment, "Role", "kops-bucket"))}"
+
+  versioning {
+    enabled = true
+  }
+}
```
Comment (marcinc) [inline s3.tf]: NL

File: variables.tf [hunk 2]
```diff
@@ -61,3 +61,8 @@ variable "master_sg_name" {
   description = "The name of the security group for the master nodes"
   default     = "masters-staged"
 }
+
+variable "kops_state_bucket" {
+  description = "The name of the state bucket to use for kops"
+  default     = ""
```
Comment (marcinc) [inline variables.tf:67]: shouldn't it have a default name?

File: variables.tf [hunk 3]
```diff
@@ -61,3 +61,8 @@ variable "master_sg_name" {
   description = "The name of the security group for the master nodes"
   default     = "masters-staged"
 }
+
+variable "kops_state_bucket" {
+  description = "The name of the state bucket to use for kops"
+  default = ""
```
Comment (gambol99) [inline variables.tf]: it will complain

File: s3.tf [hunk 4]
```diff
@@ -0,0 +1,14 @@
+# The kops state bucket
+resource "aws_s3_bucket" "kops_bucket" {
+  bucket = "${var.kops_state_bucket}"
```
Comment (gambol99) [inline s3.tf]: count = "${var.kops_state_bucket != "" ? 0 : 1}

**Pull request closed without merging**